addons/link_tracker/models/link_tracker.py

```diff
@@ -126,28 +126,27 @@ def _check_unicity(self):
                 [('campaign_id', '=', tracker.campaign_id.id)],
                 [('medium_id', '=', tracker.medium_id.id)],
                 [('source_id', '=', tracker.source_id.id)],
+                [('label', '=', tracker.label)],
             ])
             for tracker in self
         ])
 
-        # Can not be implemented with a SQL constraint because we want to care about null values.
-        all_link_trackers = self.search(search_query)
-
-        # check for unicity
-        for tracker in self:
-            if all_link_trackers.filtered(
-                lambda l: l.url == tracker.url
-                and l.campaign_id == tracker.campaign_id
-                and l.medium_id == tracker.medium_id
-                and l.source_id == tracker.source_id
-            ) != tracker:
-                raise UserError(_(
-                    'Link Tracker values (URL, campaign, medium and source) must be unique (%s, %s, %s, %s).',
-                    tracker.url,
-                    tracker.campaign_id.name,
-                    tracker.medium_id.name,
-                    tracker.source_id.name,
-                ))
+        # Can not be implemented with a SQL constraint because we care about null values.
+        potential_duplicates = self.search(search_query)
+        duplicates = []
+        seen = set()
+        for tracker in potential_duplicates:
+            unique_fields = (tracker.url, tracker.campaign_id, tracker.medium_id, tracker.source_id, tracker.label)
+            if unique_fields in seen or seen.add(unique_fields):
+                duplicates.append(tracker)
+        if duplicates:
+            error_lines = '\n- '.join([
+                str((tracker.url, tracker.campaign_id.name, tracker.medium_id.name, tracker.source_id.name, tracker.label))
+                for tracker in duplicates
+            ])
+            raise UserError(
+                _('The combination of Link Tracker values (URL, campaign, medium, source, and label) must be unique.\n'
+                  'The following combinations were already used: %(error_lines)s', error_lines=error_lines))
```

Comment:
Still voting for are :D 
```suggestion
                  'The following combinations are already used: %(error_lines)s', error_lines=error_lines))
```